pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -428,8 +428,20 @@ private Representation uploadSegment(File indexDir, File dataFile, String downlo
       throws ConfigurationException, IOException, JSONException {
     final SegmentMetadata metadata = new SegmentMetadataImpl(indexDir);
     final File tableDir = new File(baseDataDir, metadata.getTableName());
+    String tableName = metadata.getTableName();
     File segmentFile = new File(tableDir, dataFile.getName());
-    StorageQuotaChecker.QuotaCheckerResponse quotaResponse = checkStorageQuota(indexDir, metadata);
+    OfflineTableConfig offlineTableConfig = (OfflineTableConfig) ZKMetadataProvider
```

Comment:
Can you please verify that we don't hit the zk again to get the same metadata for this call once more? thanks